### DISCUSSION TITLE
If there is some API for java to load KTX2 like KTX2Loader

### DISCUSSION THREAD

**Discussion by fanbinnet** [google/filament#5712] [Q&A]
If there is some API for java to load KTX2 like KTX2Loader ?

I used the mipgen to generate ktx texture,
"mipgen --compression=etc1s blackTexture\Mat.12_BaseColor.png blackTexture\Mat.12_BaseColor.ktx2".
Then load the texture encounter a crash.
`          IoUtils.readCompressedAsset(context, "models/blackTexture/ktx/Mat.10_BaseColor.ktx2").let {
                        colorTexture = KTX1Loader.createTexture(engine, it)
                    }
`

**Comment by romainguy** [comment]
@prideout

**Reply by prideout** [reply]
Ah, currently the only way in Java is via gltfio, since we don't use them for IBL's.  I created a feature request for you:

https://github.com/google/filament/issues/5714

By the way, you probably won't need `readCompressedAsset()` because KTX2 already has a layer of lossless compression.

**Reply by fanbinnet** [reply]
Thanks for the replay.
